src/main/scala/net/psforever/actors/session/support/SessionData.scala

```diff
@@ -1334,107 +1089,115 @@ class SessionData(
-        log.info(s"${player.Name} is back")
-        player.AwayFromKeyboard = false
-        renewCharSavedTimer(
-          Config.app.game.savedMsg.renewal.fixed,
-          Config.app.game.savedMsg.renewal.variable
-        )
-      }
-      if (action == GenericActionEnum.DropSpecialItem.id) {
-        DropSpecialSlotItem()
-      }
-      if (action == 15) { //max deployment
-        log.info(s"${player.Name} has anchored ${player.Sex.pronounObject}self to the ground")
-        player.UsingSpecial = SpecialExoSuitDefinition.Mode.Anchored
-        continent.AvatarEvents ! AvatarServiceMessage(
-          continent.id,
-          AvatarAction.PlanetsideAttribute(player.GUID, 19, 1)
-        )
-        definition match {
-          case GlobalDefinitions.trhev_dualcycler | GlobalDefinitions.trhev_burster =>
-            val tool = toolOpt.get
-            tool.ToFireMode = 1
-            sendResponse(ChangeFireModeMessage(tool.GUID, 1))
-          case GlobalDefinitions.trhev_pounder =>
-            val tool = toolOpt.get
-            val convertFireModeIndex = if (tool.FireModeIndex == 0) { 1 }
-            else { 4 }
-            tool.ToFireMode = convertFireModeIndex
-            sendResponse(ChangeFireModeMessage(tool.GUID, convertFireModeIndex))
-          case _ =>
-            log.warn(s"GenericObject: ${player.Name} is a MAX with an unexpected attachment - ${definition.Name}")
-        }
-      } else if (action == 16) { //max deployment
-        log.info(s"${player.Name} has released the anchors")
-        player.UsingSpecial = SpecialExoSuitDefinition.Mode.Normal
-        continent.AvatarEvents ! AvatarServiceMessage(
-          continent.id,
-          AvatarAction.PlanetsideAttribute(player.GUID, 19, 0)
-        )
-        definition match {
-          case GlobalDefinitions.trhev_dualcycler | GlobalDefinitions.trhev_burster =>
-            val tool = toolOpt.get
-            tool.ToFireMode = 0
-            sendResponse(ChangeFireModeMessage(tool.GUID, 0))
-          case GlobalDefinitions.trhev_pounder =>
-            val tool = toolOpt.get
-            val convertFireModeIndex = if (tool.FireModeIndex == 1) { 0 }
-            else { 3 }
-            tool.ToFireMode = convertFireModeIndex
-            sendResponse(ChangeFireModeMessage(tool.GUID, convertFireModeIndex))
-          case _ =>
-            log.warn(s"GenericObject: $player is MAX with an unexpected attachment - ${definition.Name}")
-        }
-      } else if (action == 20) {
-        if (player.ExoSuit == ExoSuitType.MAX) {
-          ToggleMaxSpecialState(enable = true)
-        } else {
-          log.warn(s"GenericActionMessage: ${player.Name} can't handle action code 20")
-        }
-      } else if (action == 21) {
-        if (player.ExoSuit == ExoSuitType.MAX) {
-          player.Faction match {
-            case PlanetSideEmpire.NC =>
-              ToggleMaxSpecialState(enable = false)
+      action match {
+        case 15 => //max deployment
```

Comment:
if you have 


```scala

object Foo {
    val abc = 12
    val Bar = 24
}
```

you can then match on the upper case starting `val` : 

```scala
import Foo._

val i : Int = ???

i match {
   case Bar => ???  // valid
   case abc => ??? // not valid
}
```

that way you have a meaning to the number, because it would be hard to know what `15` might mean here.

Reply:
Clarification: I would rather create an `Enumeration` or an `EnumType` eventually.

Reply:
In scala 2.x , I was strongly discourage using `Enumeration`. It is a lot cleaner to use a `sealed trait` and `case object` . the compiler will be able to detect not exhaustive pattern matching that way

Reply:
Used `IntEnumType` `case object`s.